### PR TITLE
Coroutines Conversion: update api/html-rewriter.c++

### PR DESCRIPTION
**Pull request by jasnell**

### CODE REVIEW

File: src/workerd/api/html-rewriter.c++ [hunk 1]
```diff
@@ -446,7 +447,10 @@ kj::Promise<void> Rewriter::write(
 
 kj::Promise<void> Rewriter::end() {
   KJ_ASSERT(maybeWaitScope == nullptr);
-  return getFiberPool().startFiber([this](kj::WaitScope& scope) {
+  // We use kj::coCapture here to ensure that the this capture is maintained
```
Comment (dom96) [inline src/workerd/api/html-rewriter.c++:450]:
```suggestion
  // We use kj::coCapture here to ensure that the `this` capture is maintained
```

File: src/workerd/api/html-rewriter.c++ [hunk 2]
```diff
@@ -469,25 +474,19 @@ void Rewriter::abort(kj::Exception reason) {
...
+    co_await promise;
   }
 
-  return checkException();
+  KJ_ASSERT(writePromise == nullptr);
+
+  KJ_IF_MAYBE(exception, maybeException) {
+    inner->abort(kj::cp(*exception));
+    kj::throwFatalException(kj::cp(*exception));
+  }
```
Comment (dom96) [inline src/workerd/api/html-rewriter.c++:489]: Nicely refactored

**Review by dom96 — APPROVED** [review]
Nice, looks good

File: src/workerd/api/html-rewriter.c++ [hunk 3]
```diff
@@ -446,7 +447,10 @@ kj::Promise<void> Rewriter::write(
 
 kj::Promise<void> Rewriter::end() {
   KJ_ASSERT(maybeWaitScope == nullptr);
-  return getFiberPool().startFiber([this](kj::WaitScope& scope) {
+  // We use kj::coCapture here to ensure that the `this` capture is maintained
+  // appropriately after the co_await finishWrite()
+  return getFiberPool().startFiber(
+      kj::coCapture([this](kj::WaitScope& scope) -> kj::Promise<void> {
```
Comment (kentonv) [inline src/workerd/api/html-rewriter.c++:453]: Whoa... Here we're mixing fibers with coroutines. I think this is confusing. Only stuff up until the first `co_await` will actually run in the fiber, presumably. The `WaitScope` is only valid in the fiber.

IMO we shouldn't mix these. Let's make a rule that functions which take `kj::WaitScope&` as an input must _not_ be coroutines. We can even codify this by having the coroutine adapter fail to compile if one of the parameters is `kj::WaitScope`.

**Pull request merged**